File: Golang/inout.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	fmt.Print("Please enter your name: ")
	
	reader := bufio.NewReader(os.Stdin)
	name, _ := reader.ReadString('\n')
	
	// Trim newline character
	name = strings.TrimSpace(name)
	
	fmt.Printf("Hello, %s\n", name)
}

File: Rust/inout.rs
use std::io::{self, Write};

fn main() {
    print!("Please enter your name: ");
    // Flush to ensure the text is displayed before reading input
    io::stdout().flush().unwrap();
    
    let mut name = String::new();
    
    io::stdin()
        .read_line(&mut name)
        .expect("Failed to read line");
    
    // Remove newline character
    let name = name.trim();
    
    println!("Hello, {}", name);
}